src/repr/src/adt/mz_acl_item.rs

```diff
@@ -371,6 +376,146 @@ impl Columnation for MzAclItem {
     type InnerRegion = CloneRegion<MzAclItem>;
 }
 
+/// A list of privileges granted to a role.
+///
+/// This is primarily used for compatibility in PostgreSQL.
+///
+/// See: <https://github.com/postgres/postgres/blob/7f5b19817eaf38e70ad1153db4e644ee9456853e/src/include/utils/acl.h#L48-L59>
+#[derive(
+    Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Serialize, Hash, Deserialize, Arbitrary,
+)]
+pub struct AclItem {
+    /// Role that this item grants privileges to.
+    pub grantee: Oid,
+    /// Grantor of privileges.
+    pub grantor: Oid,
+    /// Privileges bit flag.
+    pub acl_mode: AclMode,
+}
+
+impl AclItem {
+    pub fn empty(grantee: Oid, grantor: Oid) -> AclItem {
+        AclItem {
+            grantee,
+            grantor,
+            acl_mode: AclMode::empty(),
+        }
+    }
+}
+
+// PostgreSQL has no stable binary encoding for AclItems. However, we need to be able to convert
+// types to and from binary, so we invent our own encoding. The encoding matches the in memory
+// format that PostgreSQL uses, which is (Oid, Oid, AclMode).
+impl AclItem {
+    pub fn encode_binary(&self) -> Vec<u8> {
+        let mut res = Vec::with_capacity(Self::binary_size());
+        res.extend_from_slice(&self.grantee.0.to_le_bytes());
+        res.extend_from_slice(&self.grantor.0.to_le_bytes());
+        res.extend_from_slice(&self.acl_mode.bits().to_le_bytes());
+        res
+    }
+
+    pub fn decode_binary(raw: &[u8]) -> Result<AclItem, Error> {
+        if raw.len() != AclItem::binary_size() {
+            return Err(anyhow!(
+                "invalid binary size, expecting {}, found {}",
+                AclItem::binary_size(),
+                raw.len()
+            ));
+        }
+
+        let oid_size = size_of::<u32>();
+
+        let grantee = Oid(u32::from_le_bytes(raw[0..oid_size].try_into()?));
+        let raw = &raw[oid_size..];
+        let grantor = Oid(u32::from_le_bytes(raw[0..oid_size].try_into()?));
+        let raw = &raw[oid_size..];
+        let acl_mode = u64::from_le_bytes(raw.try_into()?);
+
+        Ok(AclItem {
+            grantee,
+            grantor,
+            acl_mode: AclMode { bits: acl_mode },
+        })
+    }
```

Comment:
According to coverage data, this is covered by unit tests but not by end-to-end tests. Can we add some further tests to cover this? (Ignore if you feel that this does not make sense in this case.)

Reply:
There's no way to construct an aclitem in SQL yet, so we can't test this in end to end tests yet.